src/library_async.js

```diff
@@ -385,12 +385,14 @@ mergeInto(LibraryManager.library, {
     Asyncify.handleSleep(function(wakeUp) {
       // We must first unwind, so things are spilled to the stack. We
       // can resume right after unwinding, no need for a timeout.
-      Asyncify.afterUnwind = function() {
-        var stackBegin = Asyncify.currData + {{{ C_STRUCTS.asyncify_data_s.__size__ }}};
-        var stackEnd = HEAP32[Asyncify.currData >> 2];
-        {{{ makeDynCall('vii', 'func') }}}(stackBegin, stackEnd);
-        wakeUp();
-      };
+      new Promise(function(resolve) {
+        Asyncify.afterUnwind = function() {
+          var stackBegin = Asyncify.currData + {{{ C_STRUCTS.asyncify_data_s.__size__ }}};
+          var stackEnd = HEAP32[Asyncify.currData >> 2];
+          {{{ makeDynCall('vii', 'func') }}}(stackBegin, stackEnd);
+          resolve();
+        };
+      }).then(wakeUp());
```

Comment:
Shouldn't it looks like this?
```suggestion
      }).then(wakeUp);
```

Reply:
I removed the parentheses and updated the test case to include `emscripten_sleep`.